pkg/operator/k8sutil/node.go

```diff
@@ -278,35 +273,49 @@ func RookNodesMatchingKubernetesNodes(rookStorage rookv1.StorageScopeSpec, kuber
 	return nodes
 }
 
-func nodeTopologyLocation(kubeNode v1.Node, location string) string {
-	nodeLabels := kubeNode.ObjectMeta.Labels
-	locations := []string{location}
-
-	// We're looking for node labels that match the following format:
-	// <validTopologyLabelKey>/<key>: <value>
-	// Where validTopologyLabelKey is an entry in the
-	// validTopologyLabelKeys list, key is a topology element (e.g. region,
-	// zone), and value is the name of the element (e.g. region1, zone2)
-	for label := range nodeLabels {
-		for _, key := range validTopologyLabelKeys {
-			if strings.Contains(label, key) {
-				keys := strings.Split(label, "/")
-				locations = append(locations, fmt.Sprintf("%s=%s", keys[1], nodeLabels[label]))
-			}
-		}
+// ExtractTopologyFromLabels extracts rook topology from labels and returns a map from topology type to value
+func ExtractTopologyFromLabels(labels map[string]string, additionalTopologyIDs []string) map[string]string {
+	topology := make(map[string]string)
+
+	// check for the region k8s topology label that was deprecated in 1.17
+	const regionLabel = "region"
+	region, ok := labels[corev1.LabelZoneRegion]
+	if ok {
+		topology[regionLabel] = client.NormalizeCrushName(region)
```

Comment:
I find it a bit off that Rook SDK components are referencing Ceph components. I think something in the `operator/k8sutil` package should not normalize the labels into CRUSH formatting, and that should instead be done from the Ceph side.

Reply:
Agreed, I was intending to keep these separate. I'll make another pass at that.